pkg/kubelet/metrics/metrics.go

```diff
@@ -736,6 +737,16 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+
+	NodeStartupDuration = metrics.NewGaugeVec(
```

Comment:
This is just to make it easier to compute percentiles once metrics from all the nodes are aggregated in a monitoring platform. In Prometheus at least, you don't have any function to compute percentiles over a suite of gauge. This might be possible with some PromQL magic, but to be honest, having histograms would just make it so much simpler.

Reply:
you are the expert, I recognize my ignorance on these things 😄 , @logicalhan WDYT?

Reply:
I am not following why it is easy to aggregate a set of histogram vs a set of gauge. Google has internal tools that can easily create a histogram out of a set of gauge.

Even, with histogram, @dgrisonnet do you mean a single value in the  histogram?

Reply:
It doesn't change anything in terms of aggregation of the data, but I think that it will be easier to query the metrics. Usually, out of latency metrics you want to get the p50/p95/p99 and in Prometheus at least you can't use the `histogram_quantile` promQL function on anything other than histograms. This limitation in Prometheus is why I am suggesting to switch to a histogram here.

> Google has internal tools that can easily create a histogram out of a set of gauge.

Problem is that these tools are not available to the general audience and we want to make this metric usable by all users of Kubernetes.

> Even, with histogram, @dgrisonnet do you mean a single value in the histogram?

You won't be able to just have one value in the histogram like you have with gauge. You will have to define buckets which will make the metric lose a bit of precision.

Considering the very low cardinality of this metric, we could maybe go with one gauge and one histogram to not limit any monitoring platform?

Reply:
> Usually, out of latency metrics you want to get the p50/p95/p99 and in Prometheus at least you can't use the histogram_quantile promQL function on anything other than histograms. This limitation in Prometheus is why I am suggesting to switch to a histogram here.

> Problem is that these tools are not available to the general audience and we want to make this metric usable by all users of Kubernetes.

Yes, I agree with this. Histogram is definitely desired when people want to analyze **a set of node startup latency**.

The part I am not clear is how to aggregate single node startup values from each node into one histogram. There must be a "merging step", regardless we use gauge or histogram to represent a single node startup latency value. If you are arguing that it is straightforward to merge multiple histograms into one histogram in Prometheus (but not straightforward to merge gauges), then I will agree histogram is better.

> You won't be able to just have one value in the histogram like you have with gauge. You will have to define buckets which will make the metric lose a bit of precision.

Yes, I understand the buckets. My question is around the "count" value in a histogram. How many values we expect to collect in it.

Reply:
No worry, thank you for raising the concerns and digging into this. I think it makes more sense now because computing quantile over a 100 gauges should be easier than a 100 histograms. If the latter is supported in Prometheus, I would expect the former is supported too. 

Thank you!